test/functional/tests/failover_standby/test_failover_multihost.py

```diff
@@ -0,0 +1,534 @@
+        standby_node.cache.set_cleaning_policy(CleaningPolicy.nop)
+        standby_node.cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        TestRun.executor.run_expect_success(f"ls -la /dev/ | grep cas{cache_id}-1")
+
+
+def postfailover_check(new_primary_node, core_path, core_md5, cache_stats):
+    with TestRun.use_dut(new_primary_node), TestRun.step(f"Make sure the usage stats are correct"):
+        failover_cache_stats = new_primary_node.cache.get_statistics().usage_stats
+        if cache_stats.dirty != failover_cache_stats.dirty:
+            TestRun.LOGGER.error(
+                "The number of dirty blocks after the failover sequence doesn't match\n"
+                f"Dirty before the failover {cache_stats.dirty}\n"
+                f"Dirty after the failover {failover_cache_stats.dirty}\n"
+            )
+
+    with TestRun.use_dut(new_primary_node), TestRun.step(f"Make sure core md5 is correct"):
+        failover_core_md5 = TestRun.executor.run(f"md5sum {core_path}").stdout.split()[0]
+
+        if failover_core_md5 != core_md5:
+            TestRun.LOGGER.error("md5 after the failover sequence doesn't match")
+
+
+def new_failover_instance(new_secondary_node, drbd_resource, *, autoload):
+    if autoload:
+        with TestRun.use_dut(new_secondary_node), TestRun.step(
+            f"Verify whether the passive cache instance on {new_secondary_node.ip}"
+            f" started automatically"
+        ):
+            caches = get_caches()
+            if len(caches) < 1:
+                TestRun.LOGGER.error(f"Cache not present in system")
+            else:
+                cache_status = caches[0].get_status()
+                if cache_status != CacheStatus.standby:
+                    TestRun.LOGGER.error(
+                        f'Expected Cache state: "{CacheStatus.standby.value}" '
+                        f'Got "{cache_status.value}" instead.'
+                    )
+    else:
+        with TestRun.use_dut(new_secondary_node), TestRun.step(
+            f"Zero the standby-cache-to-be device on {new_secondary_node.ip}"
+        ):
+            dd = Dd().input("/dev/zero").output(new_secondary_node.raid.path)
+            dd.run()
+            sync()
+
+        with TestRun.use_dut(new_secondary_node), TestRun.step(
+            f"Prepare standby cache instance on {new_secondary_node.ip}"
+        ):
+            new_secondary_node.cache = casadm.standby_init(
+                cache_dev=new_secondary_node.raid,
+                cache_line_size=str(cls.value.value // 1024),
+                cache_id=cache_id,
+                force=True,
+            )
+
+    with TestRun.use_dut(new_secondary_node), TestRun.step(
+        f"Start secondary DRBD on {new_secondary_node.ip}"
+    ):
+        drbd_resource.save()
+        new_secondary_node.cache_drbd = Drbd(drbd_resource)
+        new_secondary_node.cache_drbd_dev = new_secondary_node.cache_drbd.up()
```

Comment:
In case of autoload == False (simulating new secondary node added to the cluster)  cache drbd resource must have metadata re-initialized on the new_secondary_node. I don't see it being done here.